modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3226,11 +3226,11 @@ private void processFullMessage(boolean checkCrd, ClusterNode node, GridDhtParti
                                 if (!F.isEmpty(msg.getErrorsMap())) {
                                     Exception e = msg.getErrorsMap().get(cctx.localNodeId());
 
-                                    assert e != null : msg.getErrorsMap();
+                                    if (e != null && e instanceof IgniteNeedReconnectException) {
```

Comment:
instance of operator always return `false` for `e = null`, so here we don't need check for `e != null`

Reply:
@Mmuzaf,

Fixed this.